cpp/src/io/parquet/writer_impl_helpers.hpp

```diff
@@ -0,0 +1,109 @@
+
+#include <cudf/detail/utilities/linked_column.hpp>
+#include <cudf/io/detail/parquet.hpp>
+
+namespace cudf::io::parquet::detail {
+
+/**
+ * @brief Function that translates GDF compression to parquet compression.
+ *
+ * @param compression The compression type
+ * @return The supported Parquet compression
+ */
+Compression to_parquet_compression(compression_type compression);
+
+/**
+ * @brief Function that translates the given compression codec to nvcomp compression type.
+ *
+ * @param codec Compression codec
+ * @return Translated nvcomp compression type
+ */
+nvcomp::compression_type to_nvcomp_compression_type(Compression codec);
+
+/**
+ * @brief Function that computes input alignment requirements for the given compression type.
+ *
+ * @param codec Compression codec
+ * @return Required alignment
+ */
+uint32_t page_alignment(Compression codec);
+
+/**
+ * @brief Gets the maximum compressed chunk size for the largest chunk uncompressed chunk in the
+ *        batch.
+ *
+ * @param codec Compression codec
+ * @param compression_blocksize Size of the largest uncompressed chunk in the batch
+ * @return Maximum compressed chunk size
+ */
+size_t max_compression_output_size(Compression codec, uint32_t compression_blocksize);
+
+/**
+ * @brief Fill the table metadata with default column names.
+ *
+ * @param table_meta The table metadata to fill
+ */
+void fill_table_meta(std::unique_ptr<table_input_metadata> const& table_meta);
+
+/**
+ * @brief Compute size (in bytes) of the data stored in the given column.
+ *
+ * @param column The input column
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ * @return The data size of the input
+ */
+[[nodiscard]] size_t column_size(column_view const& column, rmm::cuda_stream_view stream);
+
+/**
+ * @brief Indicates if the column should be marked as nullable in the output schema
+ *
+ * Returns `true` if the input column is nullable or if the write mode is not set to
+ * write the table all at once instead of chunked.
+ *
+ * @param column A view of the (linked) column
+ * @param column_metadata Metadata of the column
+ * @param write_mode Flag to indicate that we are guaranteeing a single table write
+ *
+ * @return Whether the column is nullable.
+ */
+[[nodiscard]] bool is_output_column_nullable(cudf::detail::LinkedColPtr const& column,
+                                             column_in_metadata const& column_metadata,
+                                             ::cudf::io::detail::single_write_mode write_mode);
+/**
+ * @brief Returns ``true`` if the given column has a fixed size.
+ *
+ * This doesn't check every row, so assumes string and list columns are not fixed, even
+ * if each row is the same width.
+ * TODO: update this if FIXED_LEN_BYTE_ARRAY is ever supported for writes.
+ *
+ * @param column A view of the column
+ *
+ * @return Whether the column has a fixed size
+ */
+[[nodiscard]] bool is_col_fixed_width(column_view const& column);
```

Comment:
Since this one isn't used for arrow, can it be left in `writer_impl.cu`? 🙏 

I'm looking at addressing the TODO since FLBA writing is supported now.

Reply:
Certainly but just to clarify, we have moved this and other helper functions to `writer_impl_helpers.cpp` to simplify `writer_impl.cu` a bit and not due to `arrow_schema_writer.cpp` in particular. Is there a particular reason to move it back to `writer_impl.cu` such as it needing CUDA code for FLBA?

Reply:
Ah, I was thinking it was a just-in-case kind of move. Never mind then, I'll just deal with the merge.